test/integration/targets/aws_dynamodb_item/tasks/insert_deletion.yml

```diff
@@ -0,0 +1,120 @@
+- name: set up aws connection info
+  set_fact:
+    aws_connection_info: &aws_connection_info
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token }}"
+      region: "{{ ec2_region }}"
+  no_log: yes
+
+- name: inserting db record
+  aws_dynamodb_item:
+    table: "{{ dynamo_db_table }}"
+    state: present
+    primary_key: {"bank": {"S": "hsbc"}, "quantity": {"N": "1000"}}
+    update_expression: "SET person =:person"
+    expression_attribute_values: {":person": {"S": "ochoa"}}
+    <<: *aws_connection_info
+  register: result
+- name: assert status
+  assert:
+    that:
+      - result.changed
+
+- name: inserting db record with a primary record that already exists
+  aws_dynamodb_item:
+    table: "{{ dynamo_db_table }}"
+    state: present
+    primary_key: {"bank": {"S": "hsbc"}, "quantity": {"N": "1000"}}
+    update_expression: "SET person =:person"
+    expression_attribute_values: {":person": {"S": "ochoa"}}
+    <<: *aws_connection_info
+  register: result
+- name: assert status
+  assert:
+    that:
+      - result.changed
```

Comment:
shouldn't this be `not result.changed` - or am I misunderstanding?

Reply:
@willthames not really. please see 

https://github.com/blastik/ansible/blob/add_dynamodb/lib/ansible/modules/cloud/amazon/aws_dynamodb_item.py#L31-L36

Reply:
With Ansible it's generally expected that things would be idempotent by default rather than only idempotent when some additional flag is set.

That's what confused me, and I'd guess willthames too.

Reply:
I think DynamoDB is an unintuitive service, especially compared with something like SimpleDB. I'm sorry this PR has been dragged out @blastik - I honestly don't think you could have chosen to add support for a much trickier service as your first contribution and I really wish I could be of more help. I've only needed to use DynamoDB a little bit, and ideally someone really familiar with both Ansible and the service could review this module to see if it feels natural. From an Ansible perspective, I understand the confusion from @willthames and @tremble and this probably needs to be a more inclusive decision, because once an interface like this is merged we won't be able to change it without a deprecation period and frustration for users.

Thanks for adding a test for idempotent behavior with `state: absent`, that part looks really good to me.

I'm okay with the way the module is achieving idempotent behavior using the conditional expression because I find dynamodb to be kind of weird and cumbersome in and of itself. But is it actually possible to get `changed: False` with `state: present` without setting `ignore_errors: true` on the task? There aren't any tests for idempotent behavior with `state: present` so I'm not sure if that works as expected. I wonder if the error code `ConditionalCheckFailedException` should result in `changed: False` instead of failure. That would be more invasive than just using the error code though, since you'd need to make sure check mode works the same way (probably with a `get_item` call). I think there should be examples and tests to help someone figure out how to get `changed: False` the second time they run a task with `state: present` even if it's dependent on a conditional expression. And I think it's a valid argument that perhaps the module should only allow overwriting values if a specific overwrite option is given (and be idempotent without a condition expression) since that's more natural to Ansible even though that's not how DynamoDB normally works.

I'm sorry for not having a clearer path forward. The module should feel natural, ideally to both DynamoDB users and Ansible users, but it seems like those things conflict and I'm not confident enough in my use of DynamoDB to feel assertive about a decision.

@jillr Do you mind taking a look at this and weighing in if you have an opinion? Maybe between the 5 of us we can come to a good solution.

Reply:
Just adding a notice that this review still stands: there should be tests and examples to show how to get `changed: false` with `state: present`.